### PR TITLE
Use yarn workspaces to generate rollup build config with correct ordering

### PR DESCRIPTION
**Pull request by jeremywiebe**
## Summary:

Rollup builds packages in the order they are returned in the build config. Right now, because we get the list of packages using essentially a directory listing, they return in alphabetical ordering, which coincidentally is our build order. But this could easily change and break. 

So this PR introduces an ordering in our Rollup config based on `yarn workspaces info` (which gives us all packages in the workspace and their in-repo depdencies). This should ensure we always build in dependency order!

Issue: "none" (but related to #171)

## Test plan:

`yarn build` orders according to dependencies

I also added a `zzz` package and then added it as a dependency on a few of the other packages in this repo. `yarn build` ordered the `zzz` package before any packages that depended on it.

### CODE REVIEW

File: config/build/rollup.config.js [hunk 1]
```diff
@@ -222,13 +292,10 @@ const getPackageInfo = (commandLineArgs, pkgName) => {
...
-const createRollupConfig = (commandLineArgs) => {
-    // Determine what packages we are building.
-    const pkgNames = fs.readdirSync("packages");
-
+const createRollupConfig = async (commandLineArgs) => {
     // For the packages we have determined we want, let's get more information
-    // about them and  generate configurations.
-    return pkgNames
+    // about them and generate configurations.
+    return getPackageDirNamesInBuildOrder()
```
Comment (jeremywiebe) [inline config/build/rollup.config.js:298]: Now that we yarn workspaces, I don't think we need to 'dir' the packages directory to figure out what to build... yarn handles that.

File: config/build/rollup.config.js [hunk 2]
```diff
@@ -34,6 +36,74 @@ const rootDir = ancesdir(__dirname);
  *               from the default behavior.
  */
 
+// Kahn's algorithm
+// https://en.wikipedia.org/wiki/Topological_sorting#Kahn's_algorithm
+const topoSort = (yarnWorkspacesOutput) => {
```
Comment (kevinbarabash) [inline config/build/rollup.config.js:41]: TIL. Thanks for including a link to the wikipedia page.

**Review by kevinbarabash — COMMENTED** [review]
LTGM.  Thanks for making the build config more robust.

File: config/build/rollup.config.js (hunk 2 above)
Comment (jeremywiebe) [reply config/build/rollup.config.js:41]: To be fair, Jared shared this implementation with me, I just updated it to work with the JSON structure that `yarn workspaces info` returns.

**Pull request merged**